linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -287,10 +298,7 @@ private final class IRChecker(unit: LinkingUnit, reporter: ErrorReporter) {
 
       case Return(expr, label) =>
         val returnType = env.returnTypes(label.name)
-        if (returnType == VoidType)
-          typecheckExpr(expr, env)
-        else
-          typecheckExpect(expr, env, returnType)
+        typecheckExpect(expr, env, returnType)
```

Comment:
This change is not correct. At least it's definitely not just a refactoring. The `expr` must be an expression even if the return type is `void`. It's a quirk of `Return`, which is also the reason that we create things like `Return(Undefined(), label)` in a number of situations.

I think it would be worth reconsidering this and expressly allow `void`s if the return type is `void`, but that's a real change that should be
* in a separate commit/PR
* accompanied by changes to replace ad hoc `Return(Undefined(), _)` by `Return(Skip(), _)` where possible, and
* perhaps changes in the backends (I don't know off the top of my head if they are prepared to receive statements in that position)

Reply:
Reverted.